mlir/lib/Dialect/QCO/Transforms/Optimizations/AuxiliaryQubitHoisting.cpp

```
@@ -0,0 +1,287 @@
+              loc, newCall.getResult(newCall.getNumResults() - 1));
+          for (unsigned i = 0; i < callOp.getNumResults(); ++i) {
+            callOp.getResult(i).replaceAllUsesWith(newCall.getResult(i));
+          }
+          callOp.erase();
+        }
+      }
+    }
+  });
+}
+
+void runAuxiliaryQubitHoisting(ModuleOp module) {
```

Cppcheck (2.21.0)

[style] The function 'runAuxiliaryQubitHoisting' is never used.

(unusedFunction)